### ISSUE TITLE
False positive for `reportUnboundVariable`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#8979]
### Discussed in https://github.com/microsoft/pyright/discussions/8978

<div type='discussions-op-text'>

<sup>Originally posted by **injust** September 12, 2024</sup>
On L9 and L11, Pyright reports that `dest` is unbound. I don't understand how this is possible.

```python
cost = at = r = 0

while True:
    if at < 0:
        cost += r - 1
        at = r

    while at != (dest := at + 1):
        r = max(r, dest)
        cost += abs(at - dest)
        at = dest
```

[Pyright Playground](https://pyright-play.net/?code=MYewzgLgBAvFCG04CdZQAwChMHcAWAlgDYCmUAKsgK4kBcmUjUBAZgtADwb1O9SiQoAahRQAtFACMDPo0Rpk2XvmJl5AQjgAKACYlBtOPKFSAlD1lRUcALbwAHluQAaKHsimZsgdBEIARmBa8hLuEJ6W7GhhmEA)</div>

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.381.
